### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by reald002**

### CODE REVIEW

File: controllers/UsersController.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            username: userData.username,
+        });
+        if(candidate){
+            const isValidPassword = bcrypt.compareSync(userData.password, candidate.password);
+            if(isValidPassword){
+                const token = jwt.sign({
+                    username: candidate.username,
+                    userId: candidate._id
+                }, process.env.JWT, {expiresIn: 60 * 60});
+                // localStorage.setItem('token', `Bearer ${token}`);
```
Comment (AnnaKulkova) [inline controllers/UsersController.js]: Don't leave comments in code

File: controllers/UsersController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+exports.register = async (req, res) => {
+    const userData = await req.body;
+    const sameName = await Users.findOne({
+        username: userData.username
+    });
+
+    if(sameName){
+        res.status(401).send(`Имя ${sameName.username} уже занято`);
```
Comment (AnnaKulkova) [inline controllers/UsersController.js]: Don't use Russian in code, please, if it is not frontend text for Russian site. And send an object, not just text (it is for all 'send').

File: models/User.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const userScheme = new Schema({
+    username: {
+        type: String,
+        required: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    // todoId: {
```
Comment (AnnaKulkova) [inline models/User.js]: Comment again

File: routes/todosRouter.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const jsonParser = bodyParser.json();
```
Comment (AnnaKulkova) [inline routes/todosRouter.js]: Don't mix require/import with usual constants. requires/imports are in the top of file, everything else is below

File: views/login.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                    password
+                })
+            })
+            const { token } = await data.json();
+            localStorage.setItem('token', token);
+
+        } catch (e) {
+            console.log(e);
+        }
+
```
Comment (AnnaKulkova) [inline views/login.html]: Try not to leave unnecessary spaces

**Review by AnnaKulkova — APPROVED** [review]

**Comment by IslamRustamov1** [comment]
**backend is ready** very weird and wrong name of the commit, do not name them like that. Commit name should contain a brief info of what has been done in present tense. For example, "create GET todos endpoint", "write tests for GET todos endpoint" and so on

**Review by IslamRustamov1 — APPROVED** [review]

**Review by IslamRustamov1 — COMMENTED** [review]
merge when ready

**Pull request merged**